### PR TITLE
Add support for OSC 1.1 tags

### LINKED ISSUE #18: parser doesn't parse T or F arguments
It looks like an argument of type 'T' or 'F' along with a few other non-standard type tags

I haven't quite got the hang of github pushing and pulling, but I think that inserting the following into F53OSCParser.h at line 260 does the job!

apologies but the formatting is a bit screwey

case 'F': //False (no data)
case 'T': //True (no data)
  numberArg = [NSNumber numberWithBool:(type=='T')];
 if ( numberArg )
 {
    [args addObject:numberArg];
    //don't advance the buffer
    if ( [[NSUserDefaults standardUserDefaults] boolForKey:@"debugIncomingOSC"] )
          NSLog( @"    bool: %@", numberArg );
  }
   else
   {
    NSLog( @"Error: Unable to parse bool argument for OSC method %@", addressPattern );
    return nil;
   }
break;

### PR DESCRIPTION
**Pull request by balord**
- Add support for sending and receiving OSC 1.1 tags `T`, `F`, `N`, and `I`
- Add support for text-to-OSC of these tags in the form of backslash-escaped `\T`, `\F`, `\N`, and `\I`
- Update F53OSCMessageTests with the new tags
- Add F53OSC_NSNumberTests to verify our assumptions about internal storage of NSNumber

Closes #18

### CODE REVIEW

File: F53OSCMessage.m [hunk 1]
```diff
@@ -321,25 +347,42 @@ - (void) setArguments:(NSArray *)argArray
...
+                    break;
                 case kCFNumberFloat32Type:
                 case kCFNumberFloat64Type:
                 case kCFNumberFloatType:
                 case kCFNumberDoubleType:
                 case kCFNumberCGFloatType:
-                    [newTypes appendString:@"f"]; break; // OSC float
+                    [newTypes appendString:@"f"]; // OSC float - 'f'
+                    break;
+                case kCFNumberCharType:
```
Comment (ChrisAshworth) [inline F53OSCMessage.m]: I've been looking for a definitive answer about how to detect boolean NSNumbers within the NSNumber [class cluster](https://developer.apple.com/library/archive/documentation/General/Conceptual/CocoaEncyclopedia/ClassClusters/ClassClusters.html), but I have had trouble finding a definitive approach.

There's a lot of discussion here (with varying degrees of reliability): https://stackoverflow.com/questions/2518761/get-type-of-nsnumber

And another review at the bottom of this page: https://nshipster.com/bool/

An example test in Stack Overflow which appears to distinguish between booleans and chars includes:

```
-(BOOL)isBool
{
    if(!strcmp(self.objCType, @encode(BOOL)) ||
        self == (void*)kCFBooleanFalse ||
        self == (void*)kCFBooleanTrue)
    {
        return YES;
    }

    if(strcmp(self.objCType, "c"))
    {
        return NO;
    }

    ....
```

Long story short, relying on `kCFNumberCharType` to signal that the NSNumber is a bool might be sufficient for F53OSC but I don't feel sure yet.

The official documentation for the NSNumber method [objCType](https://developer.apple.com/documentation/foundation/nsnumber/1807278-objctype?language=objc) suggests the comparison `!strcmp(obj.objCType, @encode(BOOL))` would be correct and perhaps more accurate, but it's not immediately clear to me why the poster on Stack Overflow included pointer comparisons to `kCFBooleanFalse` and `kCFBooleanTrue` in their "complete" answer. 

I'd love to have another dev weigh in on this too.

Comment (c-had) [reply F53OSCMessage.m]: The comparisons to `kCFBooleanFalse` and `kCFBooleanTrue` looks like a fallback to catch when the first comparison fails. `NSNumber` is toll-free bridged to `CFNumberRef` for integers and floats, and to `CFBooleanRef` for bools (see the second paragraph in the `NSNumber` doc page). It seems the reliability of `objCType` is suspect, so they added a second check that will work much of the time as well (though likely not all).

I don't really see any well-regarded, reliable way to do this either. Just a bunch of stabs that seem to work most of the time without guarantees that they always will now or in the future.

Comment (balord) [reply F53OSCMessage.m]: Yes - it seems true that we won't ever be 100% certain whether a `@YES` object in the F53OSCMessage `arguments` array can be safely transformed into an OSC `T` type.

@seandougall suggested using NSValue to represent these 4 types, and 2484a89 updates with this approach.
